src/wire/ipv4.rs

```diff
@@ -734,6 +737,17 @@ mod test {
     }
 
     #[test]
+    fn test_parse_bad_version() {
+        let mut bytes = vec![0; 24];
+        bytes.copy_from_slice(&REPR_PACKET_BYTES[..]);
+        let mut packet = Packet::new(&mut bytes);
+        packet.set_version(6);
+        packet.fill_checksum();
+        let packet = Packet::new(&*packet.into_inner());
+        assert_eq!(Repr::parse(&packet, &ChecksumCapabilities::default()), Err(Error::Malformed));
+    }
+
+    #[test]
```

Comment:
This is totally unrelated to the TTL value change. I'm 100% fine with backing this out.